### ISSUE TITLE
Allow for periods in key of radio

### ISSUE THREAD

**Issue by jrgleason** [formly-js/ng2-formly#301]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[X] feature request
[ ] support request
```

**Current behavior**
[Plunker](http://plnkr.co/edit/SN6wah4fLh0qbanSqtyM?p=preview)

If I have a key to a fieldGroup like `"key":"file.config.group.legend"` it fails but `"key":"legend"` works. The weird thing is, this only seems to be a limitation of a field Group not a field itself. For example this works....

```
[ {
  "key":"legend",
  "fieldGroup":[{
    "key":"file.rootDirectory", // notice this
    "type":"input",
    "defaultValue":null,
    "templateOptions":
    {
              "type":"text",
              "label":"file.rootDirectory.label",
              "placeholder":"file.rootDirectory.placeholder"
     }
  }]
}]
```

**Expected behavior**

Allow for periods in key of group

**Minimal reproduction of the problem with instructions**

See plunker

**What is the motivation / use case for changing the behavior?**

For namespacing this is the way our forms currently work.

P.S. I will try to look into it as well if I can get some time.

**Comment by mohammedzamakhan** [comment]
I think this got fixed by #293

**Comment by mohammedzamakhan** [comment]
In the demo application, I tried

```js
{
        key: 'address.value', // [changed]
        className: 'row',
        wrappers: ['panel'],
        templateOptions: {
          title: 'Address',
        },
        fieldGroup: [{
          className: 'col-md-6',
          type: 'input',
          key: 'street.key', // [changed]
          validators: {
            validation: ['required'],
          },
          templateOptions: {
            label: 'Street',
            placeholder: '604 Causley Ave. ',
            description: 'Enter a valid US Address',
          },
        }
}
```

and it saves it like

```js
"address": {
    "value": {
      "street": {
        "key": "input value"
      }
    }
  },
```

**Comment by jrgleason** [comment]
Sorry not sure what "Demo application" and "saves it" means. Is there a plunker with the new version so we can confirm?

I tried that on plunker and got the same results...

https://plnkr.co/edit/uidAiWaD4hVptNDd5SPA?p=preview

**Comment by jrgleason** [comment]
Sorry I think I understand, by saving it you mean what is in the ngModel right? 

That looks like it should be fine for what we are doing. It might be nice to allow people to turn that feature off if they really want the string to have a period in the key, but then again that would be invalid JSON I believe.

**Comment by mohammedzamakhan** [comment]
Right now you are using v2.0.0-beta.13, and we had a pull request after that by @Krustie101 #293 which probably resolved this issue. You cannot check this on plnkr, you have to clone this project to run the demo application. 

Clone the project, install dependencies using (yarn or npm install) and run `npm run demo` to run the demo application included in this repository.

**Comment by jrgleason** [comment]
Ok thanks I will try it out and get back to you

**Comment by jrgleason** [comment]
Another little facet to this bug is that even though the form works with a period in the field name, this causes the label to fail. So for example [here](https://plnkr.co/edit/uidAiWaD4hVptNDd5SPA?p=preview) we don't see the address label, however, removing the period in the key shows the label. Will this be fixed as well?

**Comment by mohammedzamakhan** [comment]
For now use this workaround https://plnkr.co/edit/o2gDlyQNqs3OXnwMdTzN?p=preview

**Comment by mohammedzamakhan** [comment]
Check this demo https://plnkr.co/edit/o2gDlyQNqs3OXnwMdTzN?p=preview

It uses updated formly (which is not yet released)

**Comment by jrgleason** [comment]
I just tried the demo that you mentioned and there are a couple things...

1.) Can I try this in my own app? I tried overwriting the bundle folder in node_modules but no dice.

2.) There appears to be an issue still. I changed the first couple like this...

<img width="1238" alt="screen shot 2016-12-01 at 4 09 30 pm" src="https://cloud.githubusercontent.com/assets/1319151/20812710/9b1ab65e-b7e0-11e6-9b60-6a40630ad1da.png">

I get an error...

>Cannot find control with name: 'person.title1'
